src/components/dock/dock.types.ts

```diff
@@ -0,0 +1,88 @@
+export interface DockItem {
+    /**
+     * A non-changing value to uniquely identify each item.
+     */
+    id: string;
+
+    /**
+     * Text to display for the item.
+     */
+    label: string;
+
+    /**
+     * Name of the icon to use.
+     */
+    icon: string;
+
+    /**
+     * Additional helper text for the button.
+     * Example usage can be a keyboard shortcut to activate the button.
+     */
+    helperLabel?: string;
+
+    /**
+     * Whether the button should indicate it is selected.
+     * The button should be selected when … (@Kiarokh: add description here please!)
+     */
+    selected?: boolean;
+
+    /**
+     * Used to specify a custom component to render as the dock button.
+     */
+    dockButton?: DockButton;
+
+    /**
+     * Used to specify a custom component to render as a menu for the button.
+     */
+    dockMenu?: DockMenu;
+}
+
+export interface DockButton {
+    /**
+     * Must be the tag name of a custom component implementing the
+     * [CustomDockButton](#/type/CustomDockButton/) interface.
+     */
+    componentName: string;
+
+    /**
+     * Any properties that should be set on the custom component.
+     */
+    props?: {
+        [key: string]: any;
+    };
+}
+
+export interface DockMenu {
+    /**
+     * The tag name of a custom component to be displayed in a popover when
+     * clicking on the dock item this belongs to.
+     */
+    componentName: string;
+
+    /**
+     * Any properties that should be set on the custom component.
+     */
+    props?: {
+        [key: string]: any;
+    };
+}
```

Comment:
@Befkadu1 pointed out that these two interfaces are identical, and yes, there's no reason we need two of them, I'd say. They started out different from each other, but then ended up the same, so I think we can just remove one and come up with a name for the remaining one that works for both cases.

Reply:
Although, even thought the signatures are identical, perhaps it's valuable to keep the separate documentation? It might be less confusing when looking at the docs, rather than the code… 🤔

Reply:
@Befkadu1 wrote:

> The interface `DockButton` and `DockMenu` is exactly same. The other thing I would like to say here is we might not need `dockButton?: DockButton;` for now at least. It is better to add the property when we need it. 
Having and not using it confuses the consumer. What do you think?

Reply:
Yeah, true. I think removing the option to have a custom dock-button is better than keeping it and merging the two interfaces, because I think the latter would be even more confusing.

Reply:
⚡